Light Node panics

I was running my Light Node as always, while suddenly I saw a panic. Currently have no idea what can be an issue.
![image](https://user-images.githubusercontent.com/22449364/171027236-2e1e7ce2-af54-4e0e-aa55-a833f110bfdd.png)

Same: 

```
2022-06-08T11:25:01.297Z	INFO	das	das/daser.go:239	sampled past header	{"height": 9745, "hash": "528137B14C8F9B1BFBC6DF25349758B8E9EBCEF878FEA249B75AC07EEA27C86E", "square width": 4, "finished (s)": 0.556982388}
2022-06-08T11:25:02.031Z	INFO	das	das/daser.go:239	sampled past header	{"height": 9746, "hash": "400C042F48B2E30AEDB6FE117FE9A4B2C4890681C3A7BF9CC354BCB3839E16B2", "square width": 8, "finished (s)": 0.729684527}
panic: runtime error: index out of range [0] with length 0

goroutine 26710922 [running]:
github.com/celestiaorg/celestia-node/ipld/plugin.(*namespaceHasher).Write(0x8e02f4?, {0x0?, 0x0?, 0xbf00000000000028?})
	/root/go/src/github.com/celestiaorg/celestia-node/ipld/plugin/nmt.go:94 +0x71
github.com/multiformats/go-multihash.Sum({0x0, 0x0, 0x0}, 0x1?, 0x7701?)
	/root/go/pkg/mod/github.com/multiformats/go-multihash@v0.1.0/sum.go:28 +0x77
github.com/ipfs/go-cid.Prefix.Sum({0xc019653d70?, 0x8?, 0x8?, 0xc017766980?}, {0x0?, 0xc00792cc80?, 0x484ea5?})
	/root/go/pkg/mod/github.com/ipfs/go-cid@v0.1.0/cid.go:579 +0xa5
github.com/ipfs/go-bitswap/message.newMessageFromProto({{{0x0, 0x0, 0x0}, 0x0}, {0x0, 0x0, 0x0}, {0xc014b676e0, 0x2, 0x2}, ...})
	/root/go/pkg/mod/github.com/ipfs/go-bitswap@v0.4.0/message/message.go:222 +0x1bb
github.com/ipfs/go-bitswap/message.FromMsgReader({0x7f43c4c8e998, 0xc01b37a600})
	/root/go/pkg/mod/github.com/ipfs/go-bitswap@v0.4.0/message/message.go:416 +0x118
github.com/ipfs/go-bitswap/network.(*impl).handleNewStream(0xc000b5f0e0, {0x284b7d8?, 0xc0161ebb20})
	/root/go/pkg/mod/github.com/ipfs/go-bitswap@v0.4.0/network/ipfs_impl.go:408 +0x39c
github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).SetStreamHandler.func1({0xc012149c50, 0x23}, {0x7f43c4c6e598?, 0xc0161ebb20})
	/root/go/pkg/mod/github.com/libp2p/go-libp2p@v0.15.1/p2p/host/basic/basic_host.go:529 +0x76
created by github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).newStreamHandler
	/root/go/pkg/mod/github.com/libp2p/go-libp2p@v0.15.1/p2root@renes-box:~/go/src/github.com/celestiaorg/celestia-node
```

--- a/ipld/plugin/nmt.go
+++ b/ipld/plugin/nmt.go
@@ -53,9 +53,19 @@ func NewNamespaceHasher(hasher *nmt.Hasher) hash.Hash {
 }
 
 func (n *namespaceHasher) Write(data []byte) (int, error) {
-	n.tp = data[0]
+	ln, nln, hln := len(data), int(n.NamespaceLen), n.Hash.Size()
+	innerNodeSize, leafNodeSize := (nln*2+hln)*2, nln+consts.ShareSize
+	switch ln {
+	default:
+		return 0, fmt.Errorf("wrong data size")
+	case innerNodeSize, innerNodeSize + 1: // w/ and w/o additional type byte
+		n.tp = nmt.NodePrefix
+	case leafNodeSize, leafNodeSize + 1: // w/ and w/o additional type byte
+		n.tp = nmt.LeafPrefix
+	}
+
 	n.data = data[1:]
-	return len(data), nil
+	return ln, nil
 }
 
 func (n *namespaceHasher) Sum([]byte) []byte {

